contracts/core/Vault.sol

```diff
@@ -112,11 +112,14 @@ contract Vault is Constants, ReentrancyGuard, Ownable, IVault {
     function deposit(address _account, address _token, uint256 _amount) external nonReentrant {
         uint256 collateralDeltaUsd = priceManager.tokenToUsd(_token, _amount);
         require(settingsManager.isDeposit(_token), "deposit not allowed");
-        require(
-            (settingsManager.checkDelegation(_account, msg.sender)) && _amount > 0,
-            "zero amount or not allowed for depositFor"
-        );
-        _transferIn(_account, _token, _amount);
+        require(_amount > 0, "zero amount");
+        if(_account != msg.sender) {
```

Comment:
not need to check this.
checkDelegation is processing that.
it is duplicated.

Reply:
prefer keeping it to save gas for user, as most deposit and stake txs are made by user itself